rust/main/agents/relayer/src/relayer.rs

```diff
@@ -246,18 +246,19 @@ impl BaseAgent for Relayer {
                     transaction_gas_limit
                 };
 
-            for origin in &settings.origin_chains {
+            // only iterate through origin chains that were successfully instantiated
+            for (origin, validator_annonce) in validator_announces.iter() {
                 let db = dbs.get(origin).unwrap().clone();
                 let metadata_builder = BaseMetadataBuilder::new(
                     origin.clone(),
                     destination_chain_setup.clone(),
                     prover_syncs[origin].clone(),
-                    validator_announces[origin].clone(),
+                    validator_annonce.clone(),
```

Comment:
Since we can no longer assume all `mailboxes / validator_announce` were successfully instantiated, we should loop through the successfully instantiated ones instead of blindly going through all `origin_chains / destination_chains`